docs/principles-for-package-repository-security.md

```diff
@@ -0,0 +1,105 @@
+# Principles for Package Repository Security
+
+Authors: [Jack Cable](https://github.com/cablej), [Zach Steindler](https://github.com/steiza)
+Last updated: Dec 2023
+
+## Background
+
+The Securing Software Repositories Working Group (WG) of the OpenSSF is putting together a taxonomy of package repositories and a set of principles for security capabilities. This is intended to be an opinionated document, offering a set of best practices that package repositories should strive to adhere to.
+
+By package repository (sometimes also called package registry) we're referring to the service that stores and index packages, that users and clients can download packages from. Often users and clients use a CLI tool to download packages, and some ecosystems have multiple popular CLI tools in use. Most of these security capabilities focus on the package repository, but there's also a section on security capabilities the CLI tooling can implement.
+
+The taxonomy is because not all security advice applies to all package repositories. For example, https://proxy.golang.org/ does not manage user accounts, and so has no need of documented account recovery.
+
+The roadmap of security capabilities can then be used by package repositories to assess gaps, put together fundable improvement lists ([like Python Packaging WG](https://github.com/psf/fundable-packaging-improvements/blob/master/FUNDABLES.md)) or write specific grant proposals that reference this guidance.
+
+## Taxonomy of Package Repositories
```

Comment:
Another sort of package repository is one that doesn't have user accounts and doesn't allow uploads to it directly, but only mirrors another package repository. I think it would be good to have guidance around ensuring the mirror contains the same data as the source, and perhaps guidance around how the mirror removes content that the source has removed because it's malicious?

Reply:
I sometimes differentiate between "registry" and "repository". A "repository" actually *stores* the package contents. A "registry" registers a set of packages, but depends on something else to actually store the data. Would it be useful to make this distinction?

Reply:
We settled on "package repositories" in this doc, with the clarification above:

> By package repository (sometimes also called package registry) ...

We last talked about this naming distinction back in November (!) where we looked at [how different ecosystems referred to themselves](https://docs.google.com/document/d/1VuuTnH6Hvsu_kFa_jImPaZe3-xgrn7t3KNeXBpvo5oI/edit?disco=AAABAuzRpXs), and the conclusion we came to is that there isn't consistent terminology.